### PR TITLE
add go mod check for deployments/devel and enable it in CI

### PR DESCRIPTION
**Pull request by tariq1890**

### CODE REVIEW

File: .github/workflows/golang.yaml [hunk 1]
```diff
@@ -49,7 +49,9 @@ jobs:
         args: -v --timeout 5m
         skip-cache: true
     - name: Check golang modules
-      run: make check-vendor
+      run: |
+        make check-vendor
+        make -C deployments/devel check-modules
```
Comment (elezar) [inline .github/workflows/golang.yaml:54]: Question: Why don't we use the same target in all cases?

Comment (elezar) [reply .github/workflows/golang.yaml:54]: OK. Should we then rename the target in `Makefile`?

Comment (tariq1890) [reply .github/workflows/golang.yaml:54]: Sure. So make it `check-modules` everywhere?

Comment (elezar) [reply .github/workflows/golang.yaml:54]: yes, let's do that.

Comment (tariq1890) [reply .github/workflows/golang.yaml:54]: Done, please check now

**Review by elezar — APPROVED** [review]

**Pull request merged**